### PR TITLE
PIL/Pillow image class

### PR DESCRIPTION
**Pull request by corranwebster**
This builds on #959 to provide an image class for PIL/Pillow `Image` instances.  This is part of #909 but not essential to it - we could decide to defer PIL support.

This adds an optional dependency on `pillow` - this is already a dependency of WxPython, so this is only potentially a problem for Qt users.

### CODE REVIEW

**Comment by rahulporuri** [comment]
i'm removing myself as a reviewer for now - i'll wait until after #959 is merged

File: pyface/ui_traits.py
Comment (rahulporuri) [inline pyface/ui_traits.py:49]:
```suggestion
# cache of conversions of ImageResource instances to toolkit bitmaps
```

Comment (corranwebster) [reply pyface/ui_traits.py]: Yep, caught and fixed.

**Comment by corranwebster** [comment]
This makes pillow an optional dependency:
- there is now an optional dependency for pillow
- `PILImage` is only used if imported directly - it isn't in `pyface.api`

File: pyface/i_pil_image.py
Comment (rahulporuri) [inline pyface/i_pil_image.py:10]:
```suggestion
""" The interface for a PIL Image. """
```

File: pyface/i_pil_image.py
Comment (rahulporuri) [inline pyface/i_pil_image.py:45]:
```suggestion
            The desired size as a (width, height) tuple, or None if wanting
```

File: pyface/i_pil_image.py
Comment (rahulporuri) [inline pyface/i_pil_image.py:46]: It doesn't look like the `size` argument is ignored - it gets passed through all the way down to `resize_image` from the looks of it. Isn't it?

```suggestion
            default image size.
```

File: pyface/i_pil_image.py
Comment (rahulporuri) [inline pyface/i_pil_image.py:64]: same comments here as above

```suggestion
            The desired size as a (width, height) tuple, or None if wanting
            default icon size.
```

File: pyface/pil_image.py
Comment (rahulporuri) [inline pyface/pil_image.py:11]:
```suggestion
""" The implementation of a IPILImage. """
```

File: pyface/ui/qt4/pil_image.py
Comment (rahulporuri) [inline pyface/ui/qt4/pil_image.py:11]: Tangent : ~I always wondered why these `pyface.qt` imports preceeded other imports. Why is that?~ Never mind. I think it's because the `pyface.qt` imports are technically Qt-imports which are third party packages and so they should exist before Enthought library imports.

I wish this was codified somewhere because the answer isn't obvious.

File: pyface/ui/qt4/pil_image.py
Comment (rahulporuri) [inline pyface/ui/qt4/pil_image.py:34]:
```suggestion
            The desired size as a (width, height) tuple, or None if wanting
```

File: pyface/ui/wx/pil_image.py
Comment (rahulporuri) [inline pyface/ui/wx/pil_image.py:34]:
```suggestion
            The desired size as a (width, height) tuple, or None if wanting
```

File: pyface/ui/wx/util/image_helpers.py
Comment (rahulporuri) [inline pyface/ui/wx/util/image_helpers.py:36]: Don't we normally have a new line in the function docstring between the description and the parameters?

**Review by rahulporuri — APPROVED** [review]
LGTM with a few nitpicky comments.

File: pyface/ui/qt4/pil_image.py
Comment (corranwebster) [reply pyface/ui/qt4/pil_image.py:11]: It's not done universally.  In this case it was likely just copy-pasted from `image_resource.py`.  In other cases as you note it's likely from taking Wx code and translating it and so the Wx imports get replaced by Qt imports in the same place.

It would be nice to standardize, but that would require going through and actually standardizing

File: pyface/ui/wx/util/image_helpers.py
Comment (corranwebster) [reply pyface/ui/wx/util/image_helpers.py:36]: Yes.  Not sure what happened here.  It might be a bad merge.

**Comment by corranwebster** [comment]
Right, I think this is good.  If everything passes I will merge.

**Pull request merged**
